conjure/servicewriter.go

```diff
@@ -134,13 +168,41 @@ func hasAuth(endpoints []spec.EndpointDefinition) (hasHeaderAuth, hasCookieAuth
 	return
 }
 
-func serviceInterfaceAST(serviceDefinition spec.ServiceDefinition, info types.PkgInfo, withAuth, isClient bool) (astgen.ASTDecl, StringSet, error) {
+func canAddTokenInterface(endpoints []spec.EndpointDefinition) bool {
+	numHeader := 0
+	numCookie := 0
+	for _, endpointDefinition := range endpoints {
+		if endpointDefinition.Auth == nil {
+			continue
+		}
+		possibleHeaderAuth, err := visitors.GetPossibleHeaderAuth(*endpointDefinition.Auth)
+		if err != nil {
+			return false
+		}
+		possibleCookieAuth, err := visitors.GetPossibleCookieAuth(*endpointDefinition.Auth)
+		if err != nil {
+			return false
+		}
+		if possibleHeaderAuth != nil {
+			numHeader++
+		}
+		if possibleCookieAuth != nil {
+			numCookie++
+		}
+		if numHeader != 0 && numCookie != 0 {
+			return false
```

Comment:
Why is it the case that a token interface can't be added if a service has endpoints that perform both cookie and header auth?

Reply:
There's no reason it couldn't but it's not clear to me how one token provider would be useful for generating both types of tokens. That being said, let me know if we want to change this to add a token interface on the same condition as the WithAuth interface.

Reply:
OK, makes sense.